app/scripts/lib/segment.js

```diff
@@ -20,8 +22,86 @@ const segmentNoop = {
 // provided a SEGMENT_WRITE_KEY. This also holds true for test environments and
 // E2E, which is handled in the build process by never providing the SEGMENT_WRITE_KEY
 // which process.env.IN_TEST is true
-export const segment = process.env.SEGMENT_WRITE_KEY
+const segment = process.env.SEGMENT_WRITE_KEY
   ? new Analytics(process.env.SEGMENT_WRITE_KEY, { flushAt })
   : segmentNoop
 
-export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000'
+/**
+ * Returns a function for tracking Segment events.
+ *
+ * @property {Function} getParticipateInMetrics - A function that returns
+ * whether the user participates in MetaMetrics.
+ * @property {Function} getMetaMaskVersion - A function for getting the current
+ * version of the MetaMask extension.
+ * @property {Function} getMetricsState - A function for getting state relevant
+ * to MetaMetrics/Segment.
+ */
+export function getTrackSegmentEvent (
+  getParticipateInMetrics,
+  getMetaMaskVersion,
+  getMetricsState,
+) {
+  const getVersion = () => {
```

Comment:
The version can't change at runtime, so this can be embedded directly in the context instead